shared/src/balancer_event_handler.rs

```diff
@@ -0,0 +1,154 @@
+use crate::{
+    current_block::BlockRetrieving,
+    event_handling::{BlockNumber, EventHandler, EventIndex, EventStoring},
+    impl_event_retrieving,
+    maintenance::Maintaining,
+};
+use anyhow::{anyhow, Context, Result};
+use contracts::{
+    balancer_v2_vault::{
+        self, event_data::PoolRegistered as ContractPoolRegistered, Event as ContractEvent,
+    },
+    BalancerV2Vault,
+};
+use ethcontract::common::DeploymentInformation;
+use ethcontract::{dyns::DynWeb3, Event as EthContractEvent, EventMetadata, H160};
+use std::ops::RangeInclusive;
+use tokio::sync::Mutex;
+
+#[derive(Debug)]
+pub enum BalancerEvent {
```

Comment:
Are you going to listen to other events as well in the short future? If so, which ones? If not, maybe you don't need an enum and can instead work directly with the PoolRegistered struct? (enums with one variant look a bit 🤡 - like a bullet list with one bullet point)

Reply:
I believe we will have to listen to at least one other event (`TokensDeregistered`) because apparently it is possible that the pool could become "inactive" however the Balancer team has mentioned this will likely never happen, we should at least be prepared for the fact that it can, given that it is within the functionality of the deployed contracts. Maybe @nlordell wants to say something about potential other events we might want to listen for and act upon.

Reply:
Here is a list of events emitted by the vault:

```
    AuthorizerChanged(IAuthorizer indexed newAuthorizer)
    ExternalBalanceTransfer(IERC20 indexed token, address indexed sender, address recipient, uint256 amount)
    FlashLoan(IFlashLoanRecipient indexed recipient, IERC20 indexed token, uint256 amount, uint256 feeAmount);
    InternalBalanceChanged(address indexed user, IERC20 indexed token, int256 delta)
    PausedStateChanged(bool paused);
    PoolBalanceChanged(bytes32 indexed poolId, address indexed liquidityProvider, IERC20[] tokens, int256[] deltas, uint256[] protocolFeeAmounts);
    PoolBalanceManaged(bytes32 indexed poolId,address indexed assetManager,IERC20 indexed token,int256 cashDelta,int256 managedDelta);
    PoolRegistered(bytes32 indexed poolId, address indexed poolAddress, PoolSpecialization specialization);
    RelayerApprovalChanged(address indexed relayer, address indexed sender, bool approved)
    Swap(bytes32 indexed poolId,IERC20 indexed tokenIn,IERC20 indexed tokenOut,uint256 amountIn,uint256 amountOut);
    TokensDeregistered(bytes32 indexed poolId, IERC20[] tokens);
    TokensRegistered(bytes32 indexed poolId, IERC20[] tokens, address[] assetManagers);
```

Reply:
I'm also OK with leaving it as a placeholder for now.